plugins/org.eclipse.fordiac.ide.fbtypeeditor.servicesequence/src/org/eclipse/fordiac/ide/fbtypeeditor/servicesequence/ServiceSequenceEditor.java

```diff
@@ -286,6 +285,11 @@ void setServiceSequences() {
 		if (cmds.canExecute()) {
 			cmds.execute();
 		}
+	}
 
+	@Override
+	public void setInput(final IEditorInput input) {
+		checkEditorInput(input);
```

Comment:
Should we outsource that pattern to a common base class?

something like this:

```
abstract class XY {

  @Override
  public void setInput(EditorInput) {
    checkEditorInput();
    super.setInput();
    setCustomInput();
  }
  
  abstract setCustomInput();
}
```